server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestUpdateSettingsAction.java

```diff
@@ -57,21 +57,7 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         updateSettingsRequest.setPreserveExisting(request.paramAsBoolean("preserve_existing", updateSettingsRequest.isPreserveExisting()));
         updateSettingsRequest.masterNodeTimeout(request.paramAsTime("master_timeout", updateSettingsRequest.masterNodeTimeout()));
         updateSettingsRequest.indicesOptions(IndicesOptions.fromRequest(request, updateSettingsRequest.indicesOptions()));
-
-        Map<String, Object> settings = new HashMap<>();
-        try (XContentParser parser = request.contentParser()) {
-            Map<String, Object> bodySettings = parser.map();
-            Object innerBodySettings = bodySettings.get("settings");
-            // clean up in case the body is wrapped with "settings" : { ... }
-            if (innerBodySettings instanceof Map) {
-                @SuppressWarnings("unchecked")
-                Map<String, Object> innerBodySettingsMap = (Map<String, Object>) innerBodySettings;
-                settings.putAll(innerBodySettingsMap);
-            } else {
-                settings.putAll(bodySettings);
-            }
-        }
-        updateSettingsRequest.settings(settings);
+        request.applyContentParser(parser -> updateSettingsRequest.fromXContent(parser));
```

Comment:
this makes `RequestsWithoutContentIT#testPutSettingsMissingBody` fail, as `applyContentParser` doesn't require a response body. That is only because of a different error message though, as we now fail during request validation rather than on the REST layer. I think that failing earlier was better though. Can you do `updateSettingsRequest.fromXContent(request.contentParser());` instead?